lib/mllp/client.ex

```diff
@@ -34,6 +41,67 @@ defmodule MLLP.ClientContract do
 end
 
 defmodule MLLP.Client do
+  @moduledoc """
+  MLLP.Client provides a simple tcp client for sending and receiving data i
+  via [MLLP](https://www.hl7.org/documentcenter/public/wg/inm/mllp_transport_specification.PDF) over TCP.
+
+  While MLLP is primarily used to send [HL7](https://en.wikipedia.org/wiki/Health_Level_7) messages, 
+  MLLP.Client can be used to send non-hl7 messages, such as XML.
+
+  ## Connection Behaviour 
+
+  Upon successful start up via `start_link/4`, the  client will attempt to establish a connection to the given address 
```

Comment:
There's actually a bit more to document here but I feel some changes in the future that might effect what would have been documented  (specifically, when we deem a  connection to be closed).